File: src/dotnet/SemanticKernel/Models/KnowledgeManagementCompletionRequest.cs
using FoundationaLLM.Agent.Models.Metadata;
using FoundationaLLM.Common.Models.Orchestration;
using Newtonsoft.Json;

namespace FoundationaLLM.SemanticKernel.Core.Models
{
    /// <summary>
    /// The Knowledge Management Completion Request model.
    /// </summary>
    public class KnowledgeManagementCompletionRequest : LLMCompletionRequest
    {
        /// <summary>
        /// The Knowledge Management agent metadata.
        /// </summary>
        [JsonProperty("agent")]
        public required KnowledgeManagementAgent Agent { get; set; }
    }
}

File: src/python/PythonSDK/foundationallm/models/agents/knowledge_management_completion_request.py
"""
Class Name: KnowledgeManagementCompletionRequest
Description: Encapsulates the metadata required to complete a knowledge management orchestration request.
"""
from typing import Optional
from .knowledge_management_agent import KnowledgeManagementAgent
from foundationallm.models.orchestration import CompletionRequestBase

class KnowledgeManagementCompletionRequest(CompletionRequestBase):
    """
    The completion request received from the Orchestration API.
    """    
    agent: Optional[KnowledgeManagementAgent] = None
    objects: dict = {}
